_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,865 @@
+
+Then add your GitHub username and set the new token when prompted:
+
+```shell
+ARTIFACT_ACCOUNT_NAME=your-github-username
+
+hal config artifact github account add $ARTIFACT_ACCOUNT_NAME \
+    --token
+```
+
+The new GKE, Docker, and GitHub accounts configuration must be applied to the deployment before starting the pipeline design:
+
+```shell
+hal deploy apply
+```
+
+### Create the store microservice pipeline
+
+The _pipeline_ is the central concept in deployment management with Spinnaker. It is compounded by a sequence of actions, named _stages_. A pipeline can be triggered manually or automatically, and have an execution history. Connect to the Spinnaker UI again, for creating the first pipeline.
+
+```shell
+hal deploy connect
+```
+Navigate to [localhost:9000](localhost:9000), and create the store application, choosing Kubernetes as the cloud provider. Click **Create Application** and set the following values:
+
+{% img blog/jhipster-ci-cd/sp-store-app.png alt:"Spinnaker create application" width:"550" %}{: .center-image }
+
+Then, on the left menu, choose **Pipelines** and click **Configure a new pipeline**. Set a name for the pipeline, for example, _store-cd_.
+
+### Configure the Docker image triggers
+
+In the pipeline configuration, click **Add Trigger**. Set the following configuration for the trigger:
+
+- Type: **Docker Registry**
+- Registry Name: **docker-account**
+- Organization: **your-dockerhub-username**
+- Image: **your-dockerhub-username/store**
+- Artifact Constraints: Choose **Define new artifact**
+
+In the _Expected Artifact_ form, set the following values:
+
+- Display Name: **store-image**
+- Account: **docker-registry**
+- Docker image: Fully qualified **index.docker.io/your-dockerhub-username/store**
+- Use prior execution: **yes**
+- Use default artifact: **yes** (set the same account and object path as before)
+
+Click **Save Artifact**.
+
+{% img blog/jhipster-ci-cd/sp-trigger-artifact.png alt:"Spinnaker pipeline trigger" width:"700" %}{: .center-image }
+
+**IMPORTANT NOTE**: Spinnaker provides a mechanism to [override](https://spinnaker.io/docs/guides/user/kubernetes-v2/deploy-manifest/#override-artifacts) the version of Docker images, Kubernetes ConfigMaps, and Secrets in manifests, injecting new versions when one of these objects exist in the pipeline context. However, Spinnaker documentation does not have a clear example of the trigger configuration for the docker image to be available in the pipeline context. To make it happen, you must _set it as an artifact constraint in the trigger_. If the artifact constraint is not defined, the original image version in the manifests will be deployed, instead of the new image that triggered the pipeline. If no version is in the manifests, `latest` will be the default.
```

Comment:
```suggestion
**IMPORTANT NOTE**: Spinnaker provides a mechanism to [override](https://spinnaker.io/docs/guides/user/kubernetes-v2/deploy-manifest/#override-artifacts) the version of Docker images, Kubernetes ConfigMaps, and Secrets in manifests, injecting new versions when one of these objects exists in the pipeline context. However, Spinnaker documentation does not have a clear example of the trigger configuration for the docker image to be available in the pipeline context. To make it happen, you must _set it as an artifact constraint in the trigger_. If the artifact constraint is not defined, the original image version in the manifests will be deployed, instead of the new image that triggered the pipeline. If no version is in the manifests, `latest` will be the default.
```
verb alignment ("one... exists")